cyberdrop_dl/crawlers/discord.py

```diff
@@ -0,0 +1,204 @@
+
+                if messages:
+                    timestamp = media.get("cursor", {}).get("timestamp")
+                    yield messages
+                else:
+                    break
+
+                if timestamp:
+                    request_json["tabs"]["media"]["cursor"] = {"timestamp": timestamp, "type": "timestamp"}
+
+    @error_handling_wrapper
+    async def scrape(self, scrape_item: ScrapeItem) -> None:
+        """Gets the media from the Discord mobile app search API."""
+        async for messages in self.get_media(scrape_item):
+            for message in messages:
+                await self.process_attachments(message[0], scrape_item)
+
+    async def process_attachments(self, message: dict, scrape_item: ScrapeItem) -> None:
+        for attachment in message.get("attachments"):
+            url = attachment.get("url")
+            filename = attachment.get("filename")
+            user_id = message.get("author", {}).get("id")
+            username = message.get("author", {}).get("username")
+            timestamp = self.parse_datetime(message.get("timestamp"))
+
+            canonical_url = await self.get_canonical_url(scrape_item.url)
+            if await self.check_complete_from_referer(canonical_url):
+                continue
+            new_scrape_item = scrape_item.create_new(
+                url=canonical_url,
+                new_title_part=f"{username} ({user_id})",
+                possible_datetime=timestamp,
+                add_parent=True,
+                part_of_album=True,
+            )
+
+            filename, ext = get_filename_and_ext(filename)
+            await self.handle_file(url=URL(url), scrape_item=new_scrape_item, filename=filename, ext=ext)
+            scrape_item.add_children()
+
+    @error_handling_wrapper
+    async def file(self, scrape_item: ScrapeItem) -> None:
+        """Scrapes a file."""
+        canonical_url = await self.get_canonical_url(scrape_item.url)
+        if await self.check_complete_from_referer(canonical_url):
+            return
+
+        new_scrape_item = scrape_item.create_new(url=canonical_url, add_parent=True)
+
+        filename, ext = get_filename_and_ext(scrape_item.url.name)
+        await self.handle_file(scrape_item.url, new_scrape_item, filename, ext)
+
+    """~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"""
+
+    @staticmethod
+    def parse_datetime(date: str) -> int:
+        """Parses a datetime string into a unix timestamp."""
+        date = date.split("+")[0]
+        try:
+            dt = datetime.strptime(date, "%Y-%m-%dT%H:%M:%S.%f")
+        except ValueError:
+            dt = datetime.strptime(date, "%Y-%m-%dT%H:%M:%S")
+        return calendar.timegm(dt.timetuple())
+
+    @staticmethod
+    async def get_canonical_url(url: URL) -> URL:
+        """Normalizes CDN URLs for consistency."""
+        url.with_host("cdn.discordapp.com")
+        return url
```

Comment:
This is returning the same url it got as a parameter. Should return the url with the new host.

Also, you can define this as a normal function instead of a method. It doesn't need to be async either.

```suggestion
def get_canonical_url(url: URL) -> URL:
    """Normalizes CDN URLs for consistency."""
    return url.with_host("cdn.discordapp.com")
```